integrations/gtag/lib/index.js

```diff
@@ -0,0 +1,668 @@
+    items: [getFormattedProduct(track)]
+  });
+};
+
+/**
+ * Product Added - Enhanced Ecommerce
+ *
+ * @param {Track} track
+ */
+
+GTAG.prototype.productAddedEnhanced = function(track) {
+  push('event', 'add_to_cart', {
+    items: [getFormattedProduct(track)]
+  });
+};
+
+/**
+ * Product Removed - Enhanced Ecommerce
+ *
+ * @param {Track} track
+ */
+
+GTAG.prototype.productRemovedEnhanced = function(track) {
+  push('event', 'remove_from_cart', {
+    items: [getFormattedProduct(track)]
+  });
+};
+
+/**
+ * Promotion Viewed - Enhanced Ecommerce
+ *
+ * @param {Track} track
+ */
+
+GTAG.prototype.promotionViewedEnhanced = function(track) {
+  push('event', 'view_promotion', {
+    promotions: [getFormattedPromotion(track)]
+  });
+};
+
+/**
+ * Promotion Clicked - Enhanced Ecommerce
+ *
+ * @param {Track} track
+ */
+
+GTAG.prototype.promotionClickedEnhanced = function(track) {
+  push('event', 'select_content', {
+    promotions: [getFormattedPromotion(track)]
+  });
+};
+
+/**
+ * Checkout Started - Enhanced Ecommerce
+ *
+ * @param {Track} track
+ */
+
+GTAG.prototype.checkoutStartedEnhanced = function(track) {
+  var coupon = track.proxy('properties.coupon');
+
+  push('event', 'begin_checkout', {
+    value: track.total() || track.revenue() || 0,
+    currency: track.currency(),
+    items: getFormattedProductList(track),
+    coupon: coupon
+  });
+};
+
+/**
+ * Order Updated - Enhanced Ecommerce
+ *
+ * @param {Track} track
+ */
+
+GTAG.prototype.orderUpdatedEnhanced = function(track) {
+  // Same event as started order - will override
+  this.checkoutStartedEnhanced(track);
+};
+
+/**
+ * Checkout Step Viewed - Enhanced Ecommerce
+ *
+ * @param {Track} track
+ */
+
+GTAG.prototype.checkoutStepViewedEnhanced = function(track) {
+  push('event', 'checkout_progress', extractCheckoutOptions(track));
+};
+
+/**
+ * Checkout Step Completed - Enhanced Ecommerce
+ *
+ * @param {Track} track
+ */
+
+GTAG.prototype.checkoutStepCompletedEnhanced = function(track) {
+  push('event', 'checkout_progress', extractCheckoutOptions(track));
+};
+
+/**
+ * Order Refunded - Enhanced Ecommerce
+ *
+ * @param {Track} track
+ */
+
+GTAG.prototype.orderRefundedEnhanced = function(track) {
+  var products = track.products();
+
+  var eventData = {
+    transaction_id: track.orderId()
+  };
+  if (products.length) {
+    eventData.value = track.total() || track.revenue() || 0;
+    eventData.currency = track.currency();
+    eventData.tax = track.tax();
+    eventData.shipping = track.shipping();
+    eventData.affiliation = track.properties().affiliation;
+    eventData.items = getFormattedProductList(track);
+  }
+  push('event', 'refund', eventData);
+};
+
+/**
+ * Order Completed - Enhanced Ecommerce
+ *
+ * @param {Track} track
+ */
+
+GTAG.prototype.orderCompletedEnhanced = function(track) {
+  var total = track.total() || track.revenue() || 0;
+  var orderId = track.orderId();
+  var props = track.properties();
+
+  push('event', 'purchace', {
```

Comment:
We have a typo here, this should be "purchase".

Reply:
Good catch, corrected that.